hardinfo/gpu_util.c

```diff
@@ -101,6 +101,61 @@ static void intel_fill_freq(gpud *s) {
     }
 }
 
+static void amdgpu_parse_dpmclk(gchar *path, int *min, int *max) {
+    gchar *data = NULL, *p, *next_nl;
+    int sp, i, clk;
+
+    *min = -1;
+    *max = -1;
+
+    g_file_get_contents(path, &data, NULL, NULL);
+    if (data) {
+        p = data;
+
+        while(next_nl = strchr(p, '\n')) {
```

Comment:
Won't GCC whine here because it thinks you should use `==` instead of `=`? (It stops whining if you put an extra set of parenthesis around the `while` condition.)